explorer/idl/explorer_v2.idl

```diff
@@ -0,0 +1,150 @@
+// License 2.0 that can be found in the LICENSE file.
+
+// To compile this file:
+// 1. Install the barrister translator (IDL -> JSON)
+// you need to be root (or use sudo)
+// pip install barrister
+
+// 2. Install barrister-go
+// go get github.com/coopernurse/barrister-go
+// go install github.com/coopernurse/barrister-go/idl2go
+
+// 3. barrister explorer_v2.idl | $GOPATH/bin/idl2go -i -p explorer_v2
+
+struct CoinStatistic {
+    height int
+    supply string
+    numActions int
+    aps int
+}
+
+struct BlockGenerator {
+    name string
+    address string
+}
+
+struct BlockView {
+    ID string
+    height int
+    timestamp int
+    numActions int
+    generateBy BlockGenerator
+    amount string
+    size int
+    txRoot string
+    deltaStateDigest string
+}
+
+struct Log {
+    address string
+    topics []string
+    data string
+    blockNumber int
+    txnHash string
+    blockHash string
+    index int
+}
+
+struct Receipt {
+    status int
+    hash string
+    gasConsumed int
+    contractAddress string
+    logs []Log
+}
+
+struct AddressDetails {
+    address string
+    totalBalance string
+    nonce int
+    pendingNonce int
+}
+
+struct ConsensusMetrics {
+    latestEpoch int
+    latestDelegates []string
+    latestBlockProducer string
+}
+
+struct SendActionRequest {
+    payload  string
+}
+
+struct SendActionResponse {
+    payload  string
+}
+
+struct Node {
+    address string
+}
+
+struct GetPeersResponse {
+    Self Node
+    Peers []Node
+}
+
+struct Deposit {
+    amount string
+    address string
+    confirmed bool
+}
+
+interface Explorer {
+    // get the blockchain tip height
+    getBlockchainHeight() int
+
+    // get the address detail of an address
+    getAddressDetails(address string) AddressDetails
+
+    // get list of actions by start block height, action offset and limit
+    getLastActionsByRange(startBlockHeight int, offset int, limit int) []string
+
+    // get action from action id
+    getActionByID(actionID string) string
+
+    // get list of actions belonging to an address
+    getActionsByAddress(address string, offset int, limit int) []string
+
+    // get list of unconfirmed actions in actpool belonging to an address
+    getUnconfirmedActionsByAddress(address string, offset int, limit int) []string
+
+    // get all actions in a block
+    getActionsByBlockID(blkID string, offset int, limit int) []string
+
+    // get list of blocks by block id offset and limit
+    getLastBlocksByRange(offset int, limit int) []BlockView
+
+    // get block by block id
+    getBlockByID(blkID string) BlockView
+
+    // get statistic of iotx
+    getCoinStatistic() CoinStatistic
+
+    // get consensus metrics
+    getConsensusMetrics() ConsensusMetrics
+
+    // sendAction
+    sendAction(request SendActionRequest) SendActionResponse
+
+    // get list of peers
+    getPeers() GetPeersResponse
```

Comment:
Probably we want to remove this as well because it's not user-faced API, but used for us to debug

Reply:
if you want an debug endpoint - use another interface and cover with better security check